socket_test.go

```diff
@@ -220,3 +221,37 @@ func TestConnReaperDeadlock(t *testing.T) {
 		clients[i].Close()
 	}
 }
+
+func TestSocketSendSubscriptionOnConnect(t *testing.T) {
+	endpoint := "inproc://test-resub"
+
+	sub := zmq4.NewSub(context.Background())
+	defer sub.Close()
+	pub := zmq4.NewPub(context.Background())
+	defer pub.Close()
+	sub.SetOption(zmq4.OptionSubscribe, "test")
+	if err := sub.Listen(endpoint); err != nil {
+		t.Errorf("Sub Dial failed: %v", err)
+	}
+	if err := pub.Dial(endpoint); err != nil {
+		t.Errorf("Pub Dial failed: %v", err)
+	}
+	wg := new(sync.WaitGroup)
+	defer wg.Wait()
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		time.Sleep(1 * time.Millisecond)
+		err := pub.Send(zmq4.NewMsgFromString([]string{"test"}))
+		if err != nil {
+			return
+		}
+	}()
+	msg, err := sub.Recv()
+	if err != nil {
+		t.Errorf("Recv failed: %v", err)
+	}
+	if string(msg.Frames[0]) != "test" {
+		t.Errorf("Expected 'test' got '%s'", msg.String())
```

Comment:
the Go idiom is to print "got" before "want"

Reply:
done